src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -31,6 +32,24 @@ class AutowirePass extends AbstractRecursivePass
     private $ambiguousServiceTypes = array();
     private $autowired = array();
     private $lastFailure;
+    private $throwOnAutowiringException;
+    private $autowiringExceptions = array();
```

Comment:
It's weird to have a persistent state on this class. Shouldn't it be cleared at the end of the process?